### PR TITLE
chore(tfhe): add forward compatibility with 0.6

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

**Comment by IceTDrinker** [comment]
Not expecting super in depth review given we'll change the way we handle data formats/updates

File: tfhe/c_api_tests/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ endif()
 set(TFHE_C_API_RELEASE "${CMAKE_CURRENT_SOURCE_DIR}/../../target/${CARGO_PROFILE}")
 
 option(WITH_FEATURE_GPU "Enable if tfhe-rs C API was compiled with the 'gpu' feature activated" OFF)
+option(WITH_FEATURE_GPU "Enable if tfhe-rs C API was compiled with the 'forward_compatibility' feature activated" OFF)
```
Comment (mayeul-zama) [inline tfhe/c_api_tests/CMakeLists.txt:11]: Variable name should be `WITH_FORWARD_COMPATIBILITY`

**Review by mayeul-zama — COMMENTED** [review]
Apart from a small comment, LGTM
Thanks!

File: tfhe/c_api_tests/CMakeLists.txt (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/c_api_tests/CMakeLists.txt:11]: thanks indeed 😱

File: scripts/c_api_tests.sh [hunk 2]
```diff
@@ -28,6 +31,12 @@ do
         "--gpu" )
             WITH_FEATURE_GPU="ON"
             ;;
+
+        "--forward-compat" )
+            shift
```
Comment (mayeul-zama) [inline scripts/c_api_tests.sh:36]: Is `shift` needed here?

Comment (IceTDrinker) [reply scripts/c_api_tests.sh:36]: yes it's taking an ON or OFF value in the Makefile

**Review by mayeul-zama — APPROVED** [review]
Thanks!

**Pull request merged**